src/views/Register.vue

```diff
@@ -59,13 +59,17 @@
               :placeholder="$t('general.placeholder.passwordConfirmation')"
             />
 
-            <button
+            <base-button
               type="submit"
-              class="flex justify-center w-full px-4 py-2 my-4 text-sm font-medium text-white bg-indigo-600 border border-transparent rounded-md shadow-sm hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
-              :disabled="!valid"
+              :disabled="loading === valid"
+              class="flex items-center justify-center w-full px-4 py-2 my-4 text-sm font-medium text-white border border-transparent rounded-md shadow-sm"
             >
-              {{ $t('register.button.label.signUp') }}
-            </button>
+              <i
```

Comment:
Put this in the BaseButton Component.
This state should be available on each button when the property loading is true.